### PR TITLE
Updating runejs + typescript dependencies and adding runejs/filestore

### PR DESCRIPTION
**Pull request by Tynarus**
* Removed `cache-parser` and it's uses
  * Implemented `@runejs/filestore` in it's place throughout the game engine and existing plugins
* Upgraded to TypeScript 4
* Other various dependency updates

### CODE REVIEW

File: src/plugins/objects/pickables/pickables.plugin.ts [hunk 1]
```diff
@@ -29,11 +30,11 @@ export const action: objectInteractionActionHandler = (details) => {
...
+    const pickedItem = findItem(itemId);
     setTimeout(() => {
-        details.player.sendMessage(`You ${details.option} the ${details.objectDefinition.name.toLowerCase()} and receive ${prefix} ${pickedItem.name.toLowerCase()}.`);
+        details.player.sendMessage(`You ${details.option} the ${details.objectConfig.name.toLowerCase()} and receive ${prefix} ${pickedItem.name.toLowerCase()}.`);
         details.player.playSound(2581, 7);
-        if (details.objectDefinition.name !== 'Flax' || Math.floor(Math.random() * 10) === 1) {
+        if (details.objectConfig.name !== 'Flax' || Math.floor(Math.random() * 10) === 1) {
             details.player.instance.hideGameObjectTemporarily(details.object, 30);
         }
         details.player.giveItem(pickedItem.id);
```
Comment (hallowatcher) [inline src/plugins/objects/pickables/pickables.plugin.ts]: `pickedItem.id` should be changed to `pickedItem.gameId`

**Review by hallowatcher — CHANGES_REQUESTED** [review]

**Comment by hallowatcher** [comment]
line 89 of the `gate.plugin.ts` file should be changed to `gate = hingeChunk.getFilestoreLandscapeObject(details.main, pos);`

(I cannot write a review for it inline for some reason)

image

File: src/plugins/objects/pickables/pickables.plugin.ts (hunk 1 above)
Comment (Tynarus) [reply src/plugins/objects/pickables/pickables.plugin.ts]: Fixed in the latest commit, good catch!

**Comment by hallowatcher** [comment]
There are a bunch of imports that should be updated I believe
image
image

To fix these squiggly lines in the IDE
image

**Pull request merged**